docs/proposals/local-registry-image-specs.md

```diff
@@ -0,0 +1,237 @@
+[#310](https://github.com/redhat-developer/build/issues/310).
+
+## Design Details
+
+### Test Plan
+
+1. Deploy the build operator/controller on a cluster that populates the `local-registry-hosting`
+   ConfigMap OR as cluster-admin, manually populate the `local-registry-hosting` ConfigMap.
+2. Create a `Build` and/or `BuildRun` which references a local image in the following mannter:
+   1. As output for a build.
+   2. As the `builder` image - ex. for use in the source-to-image `BuildStrategy`.
+   3. As the `runtime` base image for builds that create lean runtime images.
+3. Ensure that a build can push an image referencing the local image spec.
+
+### Graduation Criteria
+
+#### Dev Preview
+
+As an initial implementation, we can define an environment variable to set the local registry
+hostname for the build operator. This can be changed by alterting the deployment for the build
+operator, which would force a rollout with the new value.
+
+#### Tech Preview and GA
+
+The operator must fully read the `local-registry-hosting` ConfigMap, with the ability to update
+the build controller if this ConfigMap changes.
+
+#### Examples
+
+### Upgrade / Downgrade Strategy
+
+Upgrades add a new field to the API. On downgrade the `local` attribute should be ignored.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+2020-07-24: Initial proposal
+
+## Drawbacks
+
+1. The KEP is relatively new, and not many *KS providers will have a need to implement it.
+2. In production environments, may users push images to registries that reside outside of the
+   cluster. Therefore this feature may not prove valuable.
+3. `Build` and `BuildRun` definitions will not be portable across namespaces with this
+   implementation - they will only be portable across clusters.
+
+## Alternatives
+
+### okd ImageStreams
+
+OpenShift/okd [Imagestreams](https://docs.okd.io/latest/openshift_images/images-understand.html#images-imagestream-use_images-understand)
+are a primary motivation for this enhancement proposal. Shortened image pull specs which resolve to
+a local image registry is one of many features provided by this API.
+
+Upstreaming Imagestreams would require significant effort and may be beyond the scope of this
+project.
+
+### APIs for Default Search Paths
+
+Most container image build tools inject `docker.io` as the host if an image spec does not have a
+domain. This can often be overrode to reference another "default" registry:
+
+- Kaniko - the `--registry-mirror` option lets you override `docker.io` as the default. [1]
+- Buildah - uses the file `/etc/containers/registries.conf` to configure unqualified search
+  paths. [2]
+- Buildpacks - no known means of changing `docker.io` as the default path for pulling images.
+
+To reference a local image registry, the following would be needed:
+
+1. An API needs to be exposed so that the default image registry can be changed to a local registry
```

Comment:
@adambkaplan , As an alternative to exposing this as a `local:true` , have you considered having this work as an annotation?

Reply:
@sbose78 using an annotation gets us closer to ImageStreams - perhaps one of the more interesting and less documented components is the image policy resolution webhook. That takes an image ref on a pod spec and tries to resolve it to the internal registry in the form `<registry-host>/<namespace>/<spec>`